### ISSUE TITLE
"application entry point file not found Please specify in package.json" 

### ISSUE THREAD

**Issue by bharathnallamothu** [NativeScript/nativescript-angular#645]
An uncaught Exception occurred on "main" thread.
java.lang.RuntimeException: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: Application entry point file not found. Please specify the file in package.json otherwise make sure the file index.js or bootstrap.js exists.\nIf using typescript make sure your entry point file is transpiled to javascript.
 at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5406)
 at android.app.ActivityThread.-wrap2(ActivityThread.java)
 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1545)
 at android.os.Handler.dispatchMessage(Handler.java:102)
 at android.os.Looper.loop(Looper.java:154)
 at android.app.ActivityThread.main(ActivityThread.java:6119)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
Caused by: com.tns.NativeScriptException: Application entry point file not found. Please specify the file in package.json otherwise make sure the file index.js or bootstrap.js exists.\nIf using typescript make sure your entry point file is transpiled to javascript.
 at com.tns.Module.bootstrapApp(Module.java:337)
 at com.tns.Runtime.run(Runtime.java:508)
 at com.tns.NativeScriptApplication.onCreate(NativeScriptApplication.java:17)
 at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1024)
 at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5403)
 ... 8 more
Caused by: com.tns.NativeScriptException: Failed to find module: "./", relative to: app//
 at com.tns.Module.resolvePathHelper(Module.java:159)
 at com.tns.Module.bootstrapApp(Module.java:335)
 ... 12 more

**Comment by bharathnallamothu** [comment]
@tjvantoll Please help dude on above error

**Comment by tjvantoll** [comment]
@bharathnallamothu Hey, sorry you ran into issues here :( This usually happens when your `app/package.json`’s `"main"` points at a file that doesn’t exist.

As an example note how https://github.com/NativeScript/sample-Groceries/blob/282be7967aa6572100decad2c05289cfd6a3b729/app/package.json#L2 points at https://github.com/NativeScript/sample-Groceries/blob/282be7967aa6572100decad2c05289cfd6a3b729/app/main.ts.

If that doesn’t help you you follow up in our [community forum](http://forum.nativescript.org/)? (This issue tracker is for bugs in the nativescript-angular package.)

Best of luck.

**Comment by bharathnallamothu** [comment]
i got solution for above problem @tjvantoll   we have upgrade our node verion to latest (v6.9.4 LTS) 
[https://nodejs.org/dist/v6.9.4/node-v6.9.4-x64.msi](url)

**Comment by dudipsh** [comment]
Hi, 
After create a new project like below:
ng new -c=@nativescript/schematics sass-project --shared --style=scss

i got the same error when i run
tns preview

Angular CLI: 7.3.0
Node: 10.14.1
OS: win32 x64

**Comment by rodp82** [comment]
@dudipsh I had the same issue until I found this
[https://github.com/NativeScript/nativescript-angular/issues/1640](https://github.com/NativeScript/nativescript-angular/issues/1640)

**Comment by eso32** [comment]
I ran into same issue by using:
`tns priview`

solved the problem using:
`tns preview --bundle`

**Comment by Oscarato** [comment]
Uninstall previous versions

**Comment by ettolo** [comment]
Trying to get the sidedrawer TypeScript template to work in my environment, get this error when trying tns_preview
```
"Preparing project...
D:\my_path\node_modules\webpack-cli\bin\cli.js:93
Error: The entry module app specified in D:\my_path\src/package.json doesn't exist!"
```
I got the error with preview in iOS and Android

tns doctor output:
```
No issues were detected.tch (S/N)? s
√ Your ANDROID_HOME environment variable is set and points to correct directory.
√ Your adb from the Android SDK is correctly installed.
√ The Android SDK is installed.on
√ A compatible Android SDK for compilation is found.
√ Javac is installed and is configured properly.
√ The Java Development Kit (JDK) is installed and is configured properly.
√ Local builds for iOS can be executed only on a macOS system. To build for iOS on a different operating system, you can use the NativeScript cloud infrastructure.
√ Getting NativeScript components versions information...
√ Component nativescript has 7.0.10 version and is up to date.
√ Component @nativescript/core has 7.0.11 version and is up to date.
‼ Update available for component @nativescript/ios. Your current version is 7.0.1 and the latest available version is 7.0.2-rc1.
√ Component @nativescript/android has 7.0.0 version and is up to date.
```

**Comment by vatsenko** [comment]
got the same error
for me this solves problem - nativescript.config.ts

```
export default {
  id: 'org.nativescript.blank',
  appResourcesPath: 'App_Resources',
  android: {
    v8Flags: '--expose_gc',
    markingMode: 'none'
  },
  appPath: 'src',
} as NativeScriptConfig;
```
